examples/sequence_classification/p_tuning_refactor.py

```diff
@@ -0,0 +1,92 @@
+import argparse
+from transformers import (
+    AutoModelForSequenceClassification,
+    AutoTokenizer,
+    DataCollatorWithPadding,
+    TrainingArguments,
+    Trainer,
+)
+from peft import (
+    get_peft_config,
+    get_peft_model,
+    get_peft_model_state_dict,
+    set_peft_model_state_dict,
+    PeftType,
+    PromptEncoderConfig,
+)
+from datasets import load_dataset
+import evaluate
+import torch
+import numpy as np
+
+def compute_metrics(eval_pred):
+    predictions, labels = eval_pred
+    predictions = np.argmax(predictions, axis=1)
+    return metric.compute(predictions=predictions, references=labels)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Script description")
+    parser.add_argument("--model", type=str, default="roberta-large", help="Model name or path")
+    parser.add_argument("--task", type=str, default="mrpc", help="Task name")
+    parser.add_argument("--num_epochs", type=int, default=20, help="Number of epochs")
+    parser.add_argument("--lr", type=float, default=1e-3, help="Learning rate")
+    parser.add_argument("--batch_size", type=int, default=32, help="Batch size")
+    parser.add_argument("--output_dir", type=str, default="./outputs", help="Output dir")
+
+    args = parser.parse_args()
+
+    model_name_or_path = args.model
+    task = args.task
+    num_epochs = args.num_epochs
+    lr = args.lr
+    batch_size = args.batch_size
+
+    dataset = load_dataset("glue", task)
+    metric = evaluate.load("glue", task)
+
+    if any(k in model_name_or_path for k in ("gpt", "opt", "bloom")):
+        padding_side = "left"
+    else:
+        padding_side = "right"
+
+    tokenizer = AutoTokenizer.from_pretrained(model_name_or_path, padding_side=padding_side)
+    if getattr(tokenizer, "pad_token_id") is None:
+        tokenizer.pad_token_id = tokenizer.eos_token_id
+
+    def tokenize_function(examples):
+        outputs = tokenizer(examples["sentence1"], examples["sentence2"], truncation=True, max_length=None)
+        return outputs
+
+    tokenized_datasets = dataset.map(
+        tokenize_function,
+        batched=True,
+        remove_columns=["idx", "sentence1", "sentence2"],
+    )
+
+    tokenized_datasets = tokenized_datasets.rename_column("label", "labels")
+    data_collator = DataCollatorWithPadding(tokenizer=tokenizer, padding="longest")
+    peft_config = PromptEncoderConfig(task_type="SEQ_CLS", num_virtual_tokens=20, encoder_hidden_size=128)
+    model = AutoModelForSequenceClassification.from_pretrained(model_name_or_path, return_dict=True)
+    model = get_peft_model(model, peft_config)
+    training_args = TrainingArguments(
+        output_dir="your-name/roberta-large-peft-p-tuning",
```

Comment:
Since `output_dir` is hardcoded here, the `--output_dir` flag in `p_tuning_peft.sh` has no effect.

Reply:
Yes, it is fixed with parser